### PR TITLE
Use custom bond force for 1-4 interactions

### PR DESCRIPTION
**Pull request by jthorton**
## Description
This PR adds support for scaled 1-4 interactions with custom nonbonded forces. These interactions are implemented via a custom bond force between the two particles.

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [ ] TODO 1

## Questions
- [ ] Question1

## Status
- [ ] Ready to go

### CODE REVIEW

File: smirnoff_plugins/handlers/nonbonded.py [hunk 1]
```diff
@@ -58,6 +58,28 @@ def _get_potential_function(cls) -> Tuple[str, List[str], List[str]]:
...
+        Returns
+        -------
+            A string of the modified potential to be used for 1-4 interactions.
+        """
+        potential = cls._get_potential_function()[0]
+        potential = potential.split(";")
+        for i, expression in enumerate(potential):
+            if "=" not in expression:
+                # This is the final energy so modify
+                if "(" not in expression and ")" not in expression:
```
Comment (SimonBoothroyd) [inline smirnoff_plugins/handlers/nonbonded.py:76]: I wonder if we should just always add the parenthesis to be safe now that I think about it, for cases like

`a + (b * c)`, `(a + b) * c`, `(a + b) * (a + c)`

where it's hard to detect if we definitely need them?

File: smirnoff_plugins/handlers/nonbonded.py [hunk 2]
```diff
@@ -231,6 +254,43 @@ def create_force(self, system, topology, **_):
         for missing_exclusion in existing_exclusions - current_exclusions:
             force.addExclusion(*missing_exclusion)
 
+        # Add 1-4 scaled interactions only if the scale is not 1
+        if self.scale14 != 1:
```
Comment (SimonBoothroyd) [inline smirnoff_plugins/handlers/nonbonded.py:258]:
```suggestion
        if not numpy.isclose(self.scale14, 1.0):
```

File: smirnoff_plugins/tests/handlers/test_nonbonded.py [hunk 3]
```diff
@@ -82,6 +88,7 @@ def test_double_exp_energies(ideal_water_force_field):
     )
     # calculated by hand (kJ / mol), at r_min the energy should be epsilon
     ref_values = [457.0334854, -0.635968, -0.4893932627]
+    print(energies)
```
Comment (SimonBoothroyd) [inline smirnoff_plugins/tests/handlers/test_nonbonded.py:91]:
```suggestion

```

**Review by SimonBoothroyd — APPROVED** [review]
This LGTM! Thanks for adding this @jthorton !

File: smirnoff_plugins/handlers/nonbonded.py (hunk 1 above)
Comment (jthorton) [reply smirnoff_plugins/handlers/nonbonded.py:76]: Good point, that could catch us out.

**Pull request merged**